### DISCUSSION TITLE
Mons not starting properly, fail to assign global_id

### DISCUSSION THREAD

**Discussion by tomwiggers** [rook/rook#11028] [Q&A] [answered]
Hi,

I've been experimenting with setting up Rook Ceph on k3s but I cannot get it to work. I used to the Helm charts to deploy the operator and a cluster.

When I try to install a cluster (after the operator of course), Rook Ceph attempts to start 3 mons sequentially but the the first one fails to start properly (mon a). It keeps throwing errors like this one, something about it not being able to assign itself a global_id:
`debug 2022-09-20T19:55:51.348+0000 7fb2e95cd700  1 mon.a@0(probing) e3 handle_auth_request failed to assign global_id`.

I think this prevents the mon from starting properly and reaching quorum as after a while the operator seems to be out of patience, reports the following and then starts a second mon (mon b):
`2022-09-20 20:04:37.606127 E | ceph-cluster-controller: failed to reconcile CephCluster "<namespace>/<cluster-name>". failed to reconcile cluster "<cluster-name>": failed to configure local ceph cluster: failed to create cluster: failed to start ceph monitors: failed to start mon pods: failed to check mon quorum a: failed to wait for mon quorum: exceeded max retry count waiting for monitors to reach quorum`.

Mon b then does not show the same error as mon a but also does not reach quorum with mon a, which causes the operator to become impatient again and start mon c anyway. None of the mons seem to work.

It must be something simple I am overlooking but I've spent many hours on this now and cannot seem to figure it out. Until two weeks ago I had no experience with Ceph at all and never heard of Rook, so I don't know much about this.

After the operator reports the timeout I also see the following in this log, I am not sure if this is a problem or not:
`2022-09-20 20:05:42.708387 W | op-mon: failed to set Rook and/or user-defined Ceph config options before starting mons; will retry after starting mons. failed to apply default Ceph configurations: failed to set one or more Ceph configs: failed to set ceph config in the centralized mon configuration database; you may need to use the rook-config-override ConfigMap. output: Cluster connection aborted: exit status 1: failed to set ceph config in the centralized mon configuration database; you may need to use the rook-config-override ConfigMap. output: Cluster connection aborted: exit status 1: failed to set ceph config in the centralized mon configuration database; you may need to use the rook-config-override ConfigMap. output: Cluster connection aborted: exit status 1`.

Info:
k3s version: v1.24.4+k3s1
rook-ceph-operator version: v1.10.1
ceph version: 17.2.3
3 VMs, each with 2 cores, 8GB RAM and 80GB disk. (50GB empty partition for a CephBlockPool to consume).

I will be happy to provide more information but I have no idea what information is useful and what's not.

Would anybody be able to figure out what the issue is or point me in the right direction? Thanks in advance!

**Comment by tomwiggers** [comment, marked as answer]
I decided to search for solutions regarding the error about it failing to set the ceph config in the centralized mon configuration database. I stumbled upon issue [6258](https://github.com/rook/rook/issues/6258) where it's suggested to clean the `/var/lib/rook` folders on the hosts, which I did and it solved the issue.

However then the next issue began for which I also could not find a solution (and I don't see it on the troubleshooting page either :)). This one is related to the OSDs. They don't start but instead throw this error:

```
/home/jenkins-build/build/workspace/ceph-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.2.3/rpm/el8/BUILD/ceph-17.2.3/src/common/config.cc: In function 'void md_config_t::set_val_default(ConfigValues&, const ConfigTracker&, std::string_view, const string&)' thread 7fb70a8a83c0 time 2022-09-20T21:12:13.483500+0000
/home/jenkins-build/build/workspace/ceph-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.2.3/rpm/el8/BUILD/ceph-17.2.3/src/common/config.cc: 274: FAILED ceph_assert(r >= 0)
 ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x152) [0x5569011d8954]
 2: ceph-osd(+0x5d8b75) [0x5569011d8b75]
 3: (md_config_t::set_val_default(ConfigValues&, ConfigTracker const&, std::basic_string_view<char, std::char_traits<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0xc7) [0x556901a55ac7]
 4: (md_config_t::parse_env(unsigned int, ConfigValues&, ConfigTracker const&, char const*)+0x51f) [0x556901a581cf]
 5: (global_pre_init(std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const*, std::vector<char const*, std::allocator<char const*> >&, unsigned int, code_environment_t, int)+0x2e3) [0x55690199e2e3]
 6: (global_init(std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const*, std::vector<char const*, std::allocator<char const*> >&, unsigned int, code_environment_t, int, bool)+0x8a) [0x5569019a048a]
 7: main()
 8: __libc_start_main()
 9: _start()
```

This cannot be related to the `/var/lib/rook` directory on the hosts, I purged it on all 3 of them. Could it be remnants of data on the data partition (bluestore)? I used `wipefs /dev/sda2 -a`. sda2 is the 50GB partition on each host for Rook Ceph to use.

Is it better to create a new discussion for this?

**Reply by travisn** [reply]
Try the [cleanup guide](https://rook.io/docs/rook/latest/Storage-Configuration/ceph-teardown/#delete-the-data-on-hosts) to make sure all the data on the disks and the hosts are cleaned.

**Reply by tomwiggers** [reply]
Hi, thank you for answering,

I did try to use `wipefs`, `sgdisk` and `blkdiscrd` but it turns out the disks or left over data weren't the issue, I've solved it now. The OSDs weren't getting enough RAM which was apparently preventing them from starting. I'm aware the recommended amount is 2GiB (according to what the operator reports). I am now successfully running them with 1GiB but apparently less than that will cause this.

So it turns out it was my own fault for messing with the settings too much, thank you for your time to read this. I'll mark this as answer.

**Comment by pschichtel** [comment]
@travisn I think I'm having this issue after a temporary node failure in a 3 node cluster (both in terms of k8s and ceph). I have mons c, d and e and the node hosting c failed temporarily. That mon is now continuously printing `mon.c@0(probing) e6 handle_auth_request failed to assign global_id`. From the alerts I've seen in my monitoring it seems like rook briefly attempted to replace the broken mon c with a new fresh mon f, however due to the lack of additional nodes and anti-affinities that mon's pod probably couldn't be placed.

My suspicion is, that rook attempted to replace c with f and in that processes invalidated the credentials of c.

Is there any guidance on how to get the mon out of that situation? The cluster is otherwise stable.

**Reply by pschichtel** [reply]
I've also noticed that there is a mon-f in /var/lib/rook on the host of mon.c. both mon-f and mon-c have the same keyring which seems consistent with other nodes.

**Reply by pschichtel** [reply]
ok, it took quite a while, but apparently it recovered itself without intervention by me
